### PR TITLE
[MINDEXER-161] Make project Java 11

### PR DESCRIPTION
**Pull request by cstamas**
Changes:
- requires latest Java LTS to build
- requires latest Maven to build
- uses `--release` to set Java 11 level
- drop animal sniffer as JDK does all now

The goal is to keep project stick to latest LTS Java, latest Maven
and then target the required Java level using new release switch.

---

https://issues.apache.org/jira/browse/MINDEXER-161

### CODE REVIEW

**Review by slachiewicz — APPROVED** [review]

**Comment by cstamas** [comment]
@khmarbaise @gnodet Soon I'd like to make indexer a "guinea pig" to set it up as one of you mentioned on one of the Aperos:
* require latest LTS Java to build it
* require latest "stable" Maven (3.8.5 is good, IMHO should not touch it even once 3.9.0 is out, but once 4.0.1 is out, I will bump it to it)
* set release to minimal required (by libs, Lucene 9 is Java 11)

File: pom.xml [hunk 1]
```diff
@@ -84,7 +84,12 @@ under the License.
   </distributionManagement>
 
   <properties>
-    <javaVersion>8</javaVersion>
+    <javaVersion>11</javaVersion>
+    <maven.compiler.source>${javaVersion}</maven.compiler.source>
```
Comment (pzygielo) [inline pom.xml:88]: What this is for? Given that
```
error: option -source cannot be used together with --release
error: option -target cannot be used together with --release
Usage: javac <options> <source files>
use --help for a list of possible options
```
and if release is set (and in new line 359 it is) - source/target are not used by compiler-plugin anyway.

Comment (cstamas) [reply pom.xml:88]: It's in parent, so here is just redefined (as see in parent what is happening), and yes, it works as release is set... basically, just to be aligned if some tool relies on this

Comment (cstamas) [reply pom.xml:88]: See parent here https://github.com/apache/maven-parent/blob/46af4d6a5cf3b49a116e2f52e17c8343e33805e2/pom.xml#L928-L930 (this commit is parent v 36)

File: .github/workflows/maven-verify.yml [hunk 2]
```diff
@@ -25,5 +25,9 @@ jobs:
   build:
     name: Verify
     uses: apache/maven-gh-actions-shared/.github/workflows/maven-verify.yml@v2
+    with:
+      ff-jdk: '17'
+      maven_version: 3.8.5
+      jdk-matrix: '[ "17" ]'
```
Comment (slawekjaranowski) [inline .github/workflows/maven-verify.yml:31]: Why only 17?
You can  exclude only jdk 8

```
 matrix-exclude: >
        [  {"jdk": "8"} ]
```

Comment (cstamas) [reply .github/workflows/maven-verify.yml:31]: See here https://github.com/apache/maven-indexer/pull/207#issuecomment-1115316420

**Comment by jira-importer** [comment]
Resolve #592

**Pull request merged**